### ISSUE TITLE
Bacon eats the last line of output when using `bacon run`

### ISSUE THREAD

**Issue by BLucky-gh** [Canop/bacon#86]
When running `cargo run` I get this (cropped to the end)
```rust
HOST: localhost
DATABASE_URL: postgres://dev:devpwd@localhost/booru_dev
``` 
(the prompt is at the very next line with no extra newline except the one added by the prompt)
![image](https://user-images.githubusercontent.com/41377306/190406236-591fea70-873d-4184-aa6e-2c8eb0ecbc67.png)

but when running `bacon run` i get this(I have scrolled to the very bottom and scrolling more doesn't do anything, even though the scrollbar isn't at the very bottom):
```
HOST: localhost
Hit q to quit, w to wrap lines, h for help
```
![image](https://user-images.githubusercontent.com/41377306/190405707-9329c749-4a58-4d66-a212-4373b3d32937.png)

if it makes any difference: I'm running on MacOS and the terminal emulator is `kitty`. `bacon --version` returns `bacon 2.2.2`

**Comment by jamesrampton** [comment]
Same here. Tested on MacOS in kitty and iTerm2. `bacon --version` > `bacon 2.2.5`

The behaviour can be reproduced if the content fills the window, but it's easier to see in reverse mode:

Normal:
<img width="1440" alt="Screenshot 2022-10-12 at 08 57 41" src="https://user-images.githubusercontent.com/1833713/195284493-da24b9e0-8d7c-4014-bac2-55323d98bb41.png">

Reverse:
<img width="1440" alt="Screenshot 2022-10-12 at 08 56 44" src="https://user-images.githubusercontent.com/1833713/195284555-4e5411df-12d4-4273-8b50-25768f41fa09.png">

**Comment by sdrap** [comment]
I also meet this behavior when the output needs to scroll (last line is ignored).

System: ubuntu 22.04
term: default term, kitty, alacrity

with or without tmux

when the output fits without scrolling then no pb.

**Comment by Canop** [comment]
Just a note that I don't reproduce the bug, even in reverse.

**Comment by jamesrampton** [comment]
I can reproduce in Kitty, iTerm2 and Terminal.app with and without tmux. Admittedly only on MacOS Ventura (13.1).

I'd be interested to know what your environment is.

**Comment by jamesrampton** [comment]
I've narrowed this down to the extra "computing" line that's added for long-running processes like web servers.

When this line is present, the last line of output is eaten.

I added a route to my web sever which calls `std::process::exit(0)`, navigated to it, and the last line is visible again.

**Comment by Canop** [comment]
> I've narrowed this down to the extra "computing" line that's added for long-running processes like web servers.

Not sure I understand you there. You mean the task launched by `cargo run` never stops ?

**Comment by jamesrampton** [comment]
That's correct. The web server — which is the task launched by `cargo run` — will run forever under normal operation.

**Comment by Canop** [comment]
OK. I never considered watching long running tasks on bacon. But I understand it may help streamline dev (in fact I realize I could use that myself...).

I'll see what I can do for that use case.

**Comment by jamesrampton** [comment]
That's brilliant! Thank you.

**Comment by Canop** [comment]
I find it hard to reproduce, except if the last line ("Listening...") omits the new-line character at end, which would be quite impolite.

**Comment by jamesrampton** [comment]
Here's a test case that is easy to describe:

``` rust
fn main() {
    for i in 0..=10 {
        std::thread::sleep(std::time::Duration::from_millis(300));
        println!("{i}");
        println!("{}", "*".repeat(3));
    }
    println!("end");
}
```

While running with `bacon run --reverse` and no global or local `bacon.toml`, the `***` is not visible. That is to say, when i == 0, you don't see `0`'s stars. When i == 1, you then see `0`'s stars, but not `1`'s. They seem to be "hidden" behind the `computing...` line.

Additionally, I never see `end` printed, even after the program has terminated. I do, however, see `10`'s stars at this point.

**Comment by Canop** [comment]
@jamesrampton I made a branch to try fix it: https://github.com/Canop/bacon/pull/150
Could you check it and see if you still find problems ?

**Comment by jamesrampton** [comment]
LGTM @Canop, thanks!

I've tested on my usual web project and that minimal test case I posted this morning and can see the correct output lines with and without the `--reverse` flag. Have not tested thorougly elsewhere.

Just to check I'm doing it right for a fair test: I cloned the bacon repo, checked out your commit for #150 (`57ca8ea135d47781f0dde6aa73f4d901b0deb787`), built that in release mode and ran the binary directly against my two projects with `/Users/jr/code/github/bacon/target/release/bacon run --reverse`.

Hopefully that's what you were intending on me doing.
